### ISSUE TITLE
500 Internal Server Error: Could not decode to UTF-8 column 'books_author_sort'

### ISSUE THREAD

**Issue by vigri** [janeczku/calibre-web#1302]
**Describe the bug/problem**
I've a couple of books with authors having accent marks and diacritics in their name.

**Examples:**
Beljak Ján (=> Beljak, J��n)
Oždáni, Ondrej (=> O������ni, Ondrej)

There seems to be a decoding problem with calibre-web when those names are in the books_author_sort column.

**To Reproduce**
Open the main page with books, the error is showed immediately.
Opening the "backend" (/admin/view) works

**Logfile**
```
Traceback (most recent call last):
File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/engine/result.py", line 1273, in fetchall
l = self.process_rows(self._fetchall_impl())
File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/engine/result.py", line 1219, in _fetchall_impl
return self.cursor.fetchall()
sqlite3.OperationalError: Could not decode to UTF-8 column 'books_author_sort' with text 'Beljak, J��n & Neviz��nsky, Gabriel & O������ni, Ondrej & Tittonov��, Vikt��ria & Bouzek, Jan & Brez��anov��, Gabri'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 2447, in wsgi_app
response = self.full_dispatch_request()
File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1952, in full_dispatch_request
rv = self.handle_user_exception(e)
File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1821, in handle_user_exception
reraise(exc_type, exc_value, tb)
File "/usr/local/lib/python3.6/dist-packages/flask/_compat.py", line 39, in reraise
raise value
File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1950, in full_dispatch_request
rv = self.dispatch_request()
File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1936, in dispatch_request
return self.view_functions[rule.endpoint](**req.view_args)
File "/app/calibre-web/cps/web.py", line 165, in decorated_view
return func(*args, **kwargs)
File "/app/calibre-web/cps/web.py", line 478, in index
entries, random, pagination = fill_indexpage(page, db.Books, True, [db.Books.timestamp.desc()])
File "/app/calibre-web/cps/helper.py", line 774, in fill_indexpage
len(db.session.query(database).filter(db_filter).filter(common_filters()).all()))
File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/orm/query.py", line 3244, in all
return list(self)
File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/orm/loading.py", line 101, in instances
cursor.close()
File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/util/langhelpers.py", line 69, in __exit__
exc_value, with_traceback=exc_tb,
File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/util/compat.py", line 178, in raise_
raise exception
File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/orm/loading.py", line 77, in instances
fetch = cursor.fetchall()
File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/engine/result.py", line 1278, in fetchall
e, None, None, self.cursor, self.context
File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/engine/base.py", line 1482, in _handle_dbapi_exception
sqlalchemy_exception, with_traceback=exc_info[2], from_=e
File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/util/compat.py", line 178, in raise_
raise exception
File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/engine/result.py", line 1273, in fetchall
l = self.process_rows(self._fetchall_impl())
File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/engine/result.py", line 1219, in _fetchall_impl
return self.cursor.fetchall()
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) Could not decode to UTF-8 column 'books_author_sort' with text 'Beljak, J��n & Neviz��nsky, Gabriel & O������ni, Ondrej & Tittonov��, Vikt��ria & Bouzek, Jan & Brez��anov��, Gabri'
(Background on this error at: http://sqlalche.me/e/e3q8)
```

**Expected behavior**
The page with books should be displayed

**Screenshots**
![image](https://user-images.githubusercontent.com/2580723/78993209-7e469d80-7b3d-11ea-93da-3704b57887a7.png)

**Environment (please complete the following information):**
 - OS: Docker / Linux, x64
 - Calibre-Web version: latest as of today: master@4/2/20, 7:13 PM
 - Docker container: Linuxserver amd64-latest (pulled today on 10. April)
 - Special Hardware: Synology NAS (x64 CPU)
 - Browser: Firefox and Chrome

**Additional context**
Access via nginx reverse proxy

**Comment by OzzieIsaacs** [comment]
For me it's working, even with chinese characters (checked on linux and windows):
![grafik](https://user-images.githubusercontent.com/6438895/78996319-a1c11680-7b44-11ea-82ba-1065b744eb57.png)
Looks to me the content is comehow inserted wrong into the database. Would it be possible for you to run a "library check" with the Calibre program itself?  (library section and then library maintenance, library check) I don't know the exact name, as my calibre instance has a non english UI.
Do you remember how you added the books to your library? (Via calibre, or via upload on calibre-web or something else)

**Comment by vigri** [comment]
Thanks for your fast reply.
Surprisingly calibre didn't found any errors in the database.

But after examining manually the database I found a "corrupted" entry which was causing the error:

![image](https://user-images.githubusercontent.com/2580723/78997506-ffeef900-7b46-11ea-81ee-b033ae20a514.png)

Obviously there are some invalid characters  ;-)
Calibre itself just shows an empty author_sorting field for that book.

I fixed the error now by reassigning a value from calibre inside.

To be honest I don't remember if I uploaded the book through calibre or calibre-web.

Best regards
